smst_proofs_test.go

```diff
@@ -212,43 +212,57 @@ func TestSMST_ProveClosest(t *testing.T) {
 	path := sha256.Sum256([]byte("testKey2"))
 	flipPathBit(path[:], 3)
 	flipPathBit(path[:], 6)
-	closestKey, closestValueHash, closestSum, proof, err = smst.ProveClosest(path[:])
+	proof, err = smst.ProveClosest(path[:])
 	require.NoError(t, err)
-	require.NotEqual(t, proof, &SparseMerkleProof{})
-
-	result = VerifySumProof(proof, root, closestKey, closestValueHash, closestSum, NoPrehashSpec(sha256.New(), true))
-	require.True(t, result)
+	require.NotEqual(t, proof, &SparseMerkleClosestProof{})
 	closestPath := sha256.Sum256([]byte("testKey2"))
-	require.Equal(t, closestPath[:], closestKey)
-	require.Equal(t, []byte("testValue2"), closestValueHash)
-	require.Equal(t, closestSum, uint64(24))
+	closestValueHash := []byte("testValue2")
+	binary.BigEndian.PutUint64(sumBz[:], 24)
+	closestValueHash = append(closestValueHash, sumBz[:]...)
+	require.Equal(t, proof, &SparseMerkleClosestProof{
+		Path:             path[:],
+		FlippedBits:      []int{3, 6},
```

Comment:
Please use larger numbers here (and for Depth/Height) to make sure compaction doesn't break anything.

Reply:
max depth is 256 remember so still a single byte can represent it

Reply:
Again this proof compression only works for 256 bit hashers and will need to be improved over time

Reply:
Can you update the `sanity checks` to make sure the height of flips bit do not exceed the size of the bit hasher?

Reply:
they already do as Depth is checked first and they are check against it they are being 2nd hand checked against path size

Reply:
Do we have a test where the sanity check captures this and sees an error?